control-plane/connect-inject/webhook/mesh_webhook.go

```diff
@@ -311,8 +311,14 @@ func (w *MeshWebhook) Handle(ctx context.Context, req admission.Request) admissi
 			w.Log.Error(err, "error configuring injection sidecar container", "request name", req.Name)
 			return admission.Errored(http.StatusInternalServerError, fmt.Errorf("error configuring injection sidecar container: %s", err))
 		}
-		// TODO: invert to start the Envoy sidecar before the application container
-		pod.Spec.Containers = append(pod.Spec.Containers, envoySidecar)
+		//Append the Envoy sidecar before the application container only if lifecycle enabled.
+		lifecycleEnabled, ok := w.LifecycleConfig.EnableProxyLifecycle(pod)
```

Comment:
**suggestion:** You call this several times. Instead call it once before the multiport section (around line 297) and use it over and over.